programs/mpl-core/src/utils/mod.rs

```diff
@@ -292,6 +294,8 @@ pub(crate) fn validate_asset_permissions<'a>(
         }
     };
 
+    run_plugin_side_effects(&checks, Some(asset), collection, plugin_side_effect_fp)?;
```

Comment:
I could see use for a pre-side effect and post side effect for more advanced cases like if the validation step is using the data that is manipulated as part of the side effect.  Did you leave that out just to keep it simple or do you think its not needed?